server/jetstream_error.go

```diff
@@ -0,0 +1,109 @@
+package server
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/nats-io/nuid"
+)
+
+var jsLimitsExceededError = NewJSError("JS1", "A message received exceeded Stream storage limits and were dropped").UserMessage("message size exceeds maximum allowed")
+var jsStorageWriteError = NewJSError("JS2", "A message could not be stored in the JetStream storage system").UserMessage("failed to store a msg")
+var jsBadSequenceError = NewJSError("JS3", "A invalid sequence number were requested").UserMessage("bad sequence argument")
+var jsStorageReadError = NewJSError("JS4", "Reading from the JetStream storage system failed").UserMessage("could not load message from storage")
+var jsStartUpError = NewJSError("JS5", "Errors encountered during JetStream startup for an account")
```

Comment:
its annoying that these like `jsStartupError` are defined here and not in the file they are needed - but if you do it other places the code ordering is easy to mess up.  Not really a fan of this code pattern.

Previous projects I made tooling and a place on the doc side where these codes can be looked up and they hold info about what they mean and how to fix, was ok, but too much work imo.